internal/api/utils.go

```diff
@@ -370,5 +371,64 @@ func TransformComponentUnits(jsonData datatypes.JSON) map[string]interface{} {
 		}
 	}
 
+	return recommendationJSON
+}
+
+func filterNotifications(recommendationID string, clusterUUID string, recommendationJSON map[string]interface{}) map[string]interface{} {
+
+	deleteNotificationObject := func(recommendationSection map[string]interface{}) {
+		notificationObject, ok := recommendationSection["notifications"].(map[string]interface{})
+		if ok {
+			for key := range notificationObject {
+				_, found := NotificationsToShow[key]
+				if !found {
+					delete(recommendationSection, "notifications")
+					log.Warnf("dropped notification %s dropped from recommendation ID: %s; cluster ID: %s", key, recommendationID, clusterUUID)
+				}
+			}
+		}
+
+	}
```

Comment:
I would have preferred separate named func, however I can also understand this block of code is required by filterNotifications only! At the same time we can have multiple utils as per API of functionality. So I leave it upto you to keep it as is or create new func.

Reply:
Let's convert this to a named function if it gets re-used later